docs/rfcs/00-relationship-properties/relationship-properties.md

```diff
@@ -0,0 +1,341 @@
+    actorsConnection(
+        where: MovieActorWhere
+        sort: MovieActorSort
+    ): MovieActorsConnection!
+}
+```
+
+#### Input types for relationship properties
+
+Two new types will exist for each set of relationship properties - one for creating a relationship and one for updating the properties on an existing relationship. Two types are needed to respect if properties are mandatory on creation:
+
+```graphql
+input ActedInCreateInput {
+    screenTime: Int!
+}
+
+input ActedInUpdateInput {
+    screenTime: Int
+}
+```
+
+#### Changes to nested creates
+
+There are new input types to facilitate the setting of relationship properties when performing nested creates.
+
+```graphql
+input MovieCreateFieldInput {
+    properties: ActedInCreateInput!
+    node: MovieCreateInput!
+}
+
+input ActorCreateFieldInput {
+    properties: ActedInCreateInput!
+    node: ActorCreateInput!
+}
+```
+
+Now where nested create operations and `RelationInput` input types used the `ActorCreateInput` / `MovieCreateInput` types directly, they will use these in their place. This will be a breaking change and result in a major version bump.
+
+#### Changes to nested connects
+
+`ConnectFieldInput` types now contain a `properties` field for setting relationship properties on new relationships.
+
+```graphql
+input ActorConnectFieldInput {
+    where: ActorWhere
+    properties: ActedInCreateInput
+    connect: ActorConnectInput
+}
+
+input MovieConnectFieldInput {
+    where: MovieWhere
+    properties: ActedInCreateInput
+    connect: MovieConnectInput
+}
+```
+
+#### Changes to nested updates
+
+`UpdateFieldInput` types now contain a `properties` field for setting relationship properties on existing relationships.
+
+Additionally, the `where` argument will be changed so that filtering can be done on either node or relationship properties. This will be a breaking change and result in a major version bump.
+
+```graphql
+input ActorMoviesUpdateFieldInput {
+    properties: ActedInUpdateInput
+    where: ActorMovieWhere
+    update: MovieUpdateInput
+    connect: [MovieConnectFieldInput!]
+    create: [MovieCreateFieldInput!]
+    disconnect: [MovieDisconnectFieldInput!]
+    delete: [MovieDeleteFieldInput!]
+}
+
+input MovieActorsUpdateFieldInput {
+    properties: ActedInUpdateInput
+    where: MovieActorWhere
+    update: ActorUpdateInput
+    create: [ActorCreateFieldInput!]
+    connect: [ActorConnectFieldInput!]
+    disconnect: [ActorDisconnectFieldInput!]
+    delete: [ActorDeleteFieldInput!]
+}
+```
+
+### Example Queries
+
+#### With `where` and `sort` arguments
+
+For the movie "Forrest Gump", match all actors whose name begins with "Tom", and return their name and screenTime, all ordered by their name ascending.
+
+```graphql
+query {
+    movies(where: { title: "Forrest Gump" }) {
+        title
+        actorsConnection(
+            where: { node: { name_STARTS_WITH: "Tom" } }
+            sort: { node: { name: ASC } }
```

Comment:
Should we not use `options` here to keep it consistent with filtering elsewhere? Also when pagination is added then it won't need to be a breaking change.

Reply:
Compared to the page-based pagination on the current relationship fields, the arguments for cursor-based pagination (when we eventually implement it) will be based on the GraphQL Cursor Connections Specification, which has the pagination arguments at the root. 

Because of this, the `options` argument would then only contain `sort`, hence why I moved it to the top-level. However, for consistencies sake as you mention, it is probably worthwhile reintroducing the `options` argument here, and it will also allow for additional options in future if we desire.